### ISSUE TITLE
Example not working: The request signature we calculated does not match the signature you provided.

### ISSUE THREAD

**Issue by melhindi** [elerch/aws-sdk-for-zig#3]
Expected behavior: the example compiles and does not result in unexpected errors
Actual behavior: Signature mismatch reported
The used credentials work as expected with the aws-sdk for rust.

zig version: 
```
➜  example git:(master) ✗ zig version
0.13.0
```
Output:
```
➜  example git:(master) ✗ zig build run
error(aws): Call Failed

Return status: 400

Unique Request Headers:
        X-Amz-Target: TrentService.ListKeys
        Content-Type: application/x-amz-json-1.1

Request Body:
-------------
{
    "Limit": null,
    "Marker": null
}
-------------
Response Headers:
        x-amzn-RequestId: 6f52b8f9-646d-4bdd-81f1-b19d80bcd2a6
        Cache-Control: no-cache, no-store, must-revalidate, private
        Expires: 0
        Pragma: no-cache
        Date: Thu, 08 Aug 2024 16:15:03 GMT
        Content-Type: application/x-amz-json-1.1
        Content-Length: 229
        connection: keep-alive
Response Body:
--------------
{"__type":"InvalidSignatureException","message":"The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details."}
--------------


Calling KMS ListKeys, a TLS 1.3 enabled service
You likely have at least some AWS-generated keys in your account,
but if the account has not had many services used, this may return 0 keys

error: HttpFailure
/home/melhindi/.cache/zig/p/122015b35d150e9c641a8f577ea07bb5cf52020038d052bcabc5d89214ac36e82c4e/src/aws.zig:275:17: 0x116fb3a in callAws (tmp)
                return error.HttpFailure;
                ^
/home/melhindi/.cache/zig/p/122015b35d150e9c641a8f577ea07bb5cf52020038d052bcabc5d89214ac36e82c4e/src/aws.zig:212:20: 0x1170261 in callJson (tmp)
            return try Self.callAws(.{
                   ^
/home/melhindi/.cache/zig/p/122015b35d150e9c641a8f577ea07bb5cf52020038d052bcabc5d89214ac36e82c4e/src/aws.zig:96:41: 0x1170623 in call (tmp)
                .json_1_0, .json_1_1 => return Self.callJson(request, options),
                                        ^
/home/melhindi/git/aws-sdk-for-zig/example/src/main.zig:45:22: 0x113039d in main (tmp)
    const call_kms = try aws.Request(services.kms.list_keys).call(.{}, options);
                     ^
run
└─ run tmp failure
error: the following build command failed with exit code 1:
/home/melhindi/git/aws-sdk-for-zig/example/.zig-cache/o/45d9971f039f34fc2dc15b220b61b211/build /home/melhindi/.zigenv/versions/0.13.0/zig /home/melhindi/git/aws-sdk-for-zig/example /home/melhindi/git/aws-sdk-for-zig/example/.zig-cache /home/melhindi/.cache/zig --seed 0xad317052 -Z584b98b2b6ffb4b5 run
```

**Comment by elerch** [comment]
I will take a look  - what authentication mechanism are you using?

**Comment by elerch** [comment]
FYI - I was able to repro locally and will look, but auth method may still be useful.

**Comment by melhindi** [comment]
Hi, thanks for looking into this:
I used account access_keys in the credentials file to avoid any issues with permissions.

**Comment by elerch** [comment]
The root cause of this is #4. Both should be resolved by  e5b662873a6745a7e761643b1ca3d8637bf1222f.

I'll close this one, and keep the other open for the time being
